homeassistant/components/advantage_air/climate.py

```diff
@@ -166,22 +166,19 @@ def __init__(self, instance, ac_key, zone_key):
             f'{self.coordinator.data["system"]["rid"]}-{ac_key}-{zone_key}'
         )
 
-    @property
-    def current_temperature(self):
-        """Return the current temperature."""
-        return self._zone["measuredTemp"]
-
-    @property
-    def target_temperature(self):
-        """Return the target temperature."""
-        return self._zone["setTemp"]
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        self.async_on_remove(self.coordinator.async_add_listener(self._update_callback))
 
-    @property
-    def hvac_mode(self):
-        """Return the current HVAC modes."""
+    @callback
+    def _update_callback(self) -> None:
+        """Load data from integration."""
+        self._attr_current_temperature = self._zone["measuredTemp"]
+        self._attr_target_temperature = self._zone["setTemp"]
+        self._attr_hvac_mode = HVAC_MODE_OFF
         if self._zone["state"] == ADVANTAGE_AIR_STATE_OPEN:
-            return HVAC_MODE_FAN_ONLY
-        return HVAC_MODE_OFF
+            self._attr_hvac_mode = HVAC_MODE_FAN_ONLY
```

Comment:
```suggestion
            self._attr_hvac_mode = HVAC_MODE_AUTO
```
To help merge witth https://github.com/home-assistant/core/pull/52816